staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -118,6 +133,24 @@ type APIAggregator struct {
 	// handledGroups are the groups that already have routes
 	handledGroups sets.String
 
+	// Swagger spec for each api service
+	apiServiceSpecs map[string]*spec.Swagger
```

Comment:
These should all be abstracted into an interface that accepts Add/Remove apiServices and the logic should be contained close to the handler you're coding.